### PR TITLE
Action tests + login cleanup

### PR DESCRIPTION
**Pull request by toddmedema**

### CODE REVIEW

File: app/actions/User.tsx [hunk 1]
```diff
@@ -1,94 +1,97 @@
...
 import {UserState} from '../reducers/StateTypes'
 import {authSettings} from '../Constants'
 
 declare var gapi: any;
 declare var window: any;
 
 
-function registerUserAndIdToken(user: {name: string, image: string, email: string}, idToken: string, cb: (user:UserState) => any) {
-  var xhr = new XMLHttpRequest();
+function registerUserAndIdToken(user: {name: string, image: string, email: string}, idToken: string, callback: (user:UserState, err?: string) => any) {
```
Comment (smartin015) [inline app/actions/User.tsx]: We could actually `type UserLoginCallback = (user: UserState, err?: string) => any;`, which would give us a standard callback interface for all of these things (and reduce redundant code).

File: app/actions/User.tsx [hunk 2]
```diff
@@ -122,17 +128,17 @@ export function logout() {
 }
 
 /*
```
Comment (smartin015) [inline app/actions/User.tsx]: hmmmmmmmmm....

File: app/actions/User.tsx (hunk 1 above)
Comment (toddmedema) [inline app/actions/User.tsx]: Good idea

File: app/actions/User.tsx (hunk 2 above)
Comment (toddmedema) [inline app/actions/User.tsx]: Getting logout to work's outside of the scope of this PR, but I'd bet that the commented-out code'll be pretty close to the final solution / don't want to delete it

Comment (smartin015) [inline app/actions/User.tsx]: Fair 'nuff.

**Pull request merged**